packages/cardano-services/src/util/DbSyncProvider/DbSyncProvider.ts

```diff
@@ -26,27 +48,37 @@ export const DbSyncProvider = <
   BaseClass?: T
 ) => {
   abstract class Mixin extends (BaseClass || Object) implements Provider {
-    public db: Pool;
+    public dbPools: Required<DbPools>;
     public cardanoNode: CardanoNode;
     public logger: Logger;
+    #cache: {
+      healthCheck: InMemoryCache;
+    };
 
     constructor(...args: AnyArgs) {
       const [dependencies, ...baseArgs] = [...args] as [DbSyncProviderDependencies, ...AnyArgs];
-      const { db, cardanoNode, logger } = dependencies;
-
+      const { cache, dbPools, cardanoNode, logger } = dependencies;
       super(...baseArgs);
 
-      this.db = db;
+      this.dbPools = dbPools;
       this.cardanoNode = cardanoNode;
       this.logger = logger;
+      this.#cache = cache?.healthCheck
```

Comment:
This can be simplified once made required the cache engine

Reply:
I think the intention for this to be optional was to facilitate our tests preps, so the `cache` creation can be skipped there. As in the [providerServer](https://github.com/input-output-hk/cardano-js-sdk/blob/1f1a2c83a1c8208ddb7dd2d6decc88c74a1f719b/packages/cardano-services/src/Program/programs/providerServer.ts#L82-L83) impl we always create the shared `healthCheckCache`.
Going to make it required @iccicci

Reply:
426169643da6936df4ff1d85b19c29fc897c9135